src/plugins/immutable-properties.ts

```diff
@@ -187,10 +200,18 @@ function shouldForwardDeclareAttribute(valueTypeName:string, makePublicImports:b
   return declaringPublicAttributes || attributeTypeReferencesObjectType;
 }
 
-function forwardDeclarationForAttribute(attribute:ValueObject.Attribute): ObjC.ForwardDeclaration {
+function forwardClassDeclarationForAttribute(attribute:ValueObject.Attribute): ObjC.ForwardDeclaration {
   return ObjC.ForwardDeclaration.ForwardClassDeclaration(attribute.type.name);
 }
 
+function forwardProtocolDeclarationForAttribute(attribute:ValueObject.Attribute): ObjC.ForwardDeclaration {
+  return Maybe.match(function (protocol) {
+    return ObjC.ForwardDeclaration.ForwardProtocolDeclaration(protocol);
+  }, function () {
+    return undefined;
```

Comment:
The function returns an `ObjC.ForwardDeclaration` but you're returning undefined here. The types need to be updated (I think) if you're planning on returning undefined